packages/expo-constants/scripts/get-app-config-ios.sh

```diff
@@ -8,10 +8,18 @@ RESOURCE_BUNDLE_NAME="EXConstants.bundle"
 # Path to expo-constants folder inside node_modules
 EXPO_CONSTANTS_PACKAGE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
 
-# Suppress environment errors from sourcing the login scripts
-set +e
-source "$EXPO_CONSTANTS_PACKAGE_DIR/scripts/source-login-scripts.sh"
-set -e
+
+export NVM_DIR=$HOME/.nvm;
+NVM_SH_FILE=$NVM_DIR/nvm.sh
+# first attempt to source via nvm and fallback to sourcing via nvm
+if test -f "$NVM_SH_FILE"; then
+  source $NVM_SH_FILE
+else
+  # Suppress environment errors from sourcing the login scripts
+  set +e
+  source "$EXPO_CONSTANTS_PACKAGE_DIR/scripts/source-login-scripts.sh"
```

Comment:
does expo rely on this login sourcing for anything other than finding the node binary? 

I have been testing this fix with teammates who use nvm and it seems to work well.

Reply:
if this approach makes sense, then I can move this logic to this `source-login-scripts.sh` (and rename that to something like `source-node-binary`).

Reply:
Have you tested with and nvm and zsh? I think that's when the problem occurs

Reply:
Yes - that is my main use case and reason for submitting here. My team uses zsh and nvm - we've had a ton of issues with the script as is, and this approach resolves things for us